Google.Api.Gax.Grpc/Rest/HttpRuleTranscoder.cs

```diff
@@ -268,46 +268,46 @@ internal IEnumerable<string> GetValues(IMessage request)
                 // int32, sint32, sfixed32, int64, sint64, sfixed64
                 value is 0 || value is 0L ||
                 // uint32, fixed32, uint64, fixed64
-                value is UnsignedInt32Zero || value is 0UL || 
+                value is UnsignedInt32Zero || value is 0UL ||
                 // bytes
                 (value is ByteString bs && bs.IsEmpty) ||
                 // Any enum (all protobuf enums have an underlying type of Int32)
                 (value is Enum enumValue && Convert.ToInt32(enumValue) == 0);
-
-            static string FormatValue(object value) => 
-                value is bool b ? (b ? "true" : "false")
-                : value is Enum enumValue ? OriginalEnumValueHelper.GetOriginalName(enumValue)
-                : value is IFormattable formattable ? formattable.ToString(format: null, CultureInfo.InvariantCulture)
-                : value.ToString();
         }
+    }
 
-        // Effectively a cache of mapping from enum values to the original name as specified in the proto file,
-        // fetched by reflection. This code is taken from the protobuf JsonFormatter.
-        private static class OriginalEnumValueHelper
-        {
-            private static readonly ConcurrentDictionary<System.Type, Dictionary<object, string>> _dictionaries
-                = new ConcurrentDictionary<System.Type, Dictionary<object, string>>();
+    internal static string FormatPathOrQueryValue(object value) =>
+        value is bool b ? (b ? "true" : "false")
+        : value is Enum enumValue ? OriginalEnumValueHelper.GetOriginalName(enumValue)
+        : value is IFormattable formattable ? formattable.ToString(format: null, CultureInfo.InvariantCulture)
```

Comment:
For the floating point numbers, is everyone else using invariant culture, or server side at least? Is that one of the things that's unclear?

Reply:
Everyone should definitely be using the invariant culture or equivalent (no "1,5" for example) but there are other problems around precision. Happy to talk about it on a VC if that would be useful.